internal/cmd/root.go

```diff
@@ -0,0 +1,28 @@
+// Copyright The Envoy Project Authors
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+// 	http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"github.com/spf13/cobra"
+)
+
+// GetRootCommand returns the root cobra command to be executed
+// by main.
+func GetRootCommand() *cobra.Command {
+	return &cobra.Command{
+		Use:   "envoy-gateway",
```

Comment:
`envoy-gateway` is a lot to type, but I can't think of anything better rn,
will the server/daemon CLI (`envoy-gateway`) be different that the client CLI ( which I envision will be required for debuggability) or will they be subcommands within `envoy-gateway`  (I'd vote for the latter) ?

Reply:
Yeah, `envoy-gateway` seemed better than `gateway`, but it can certainly change to something else if we're not happy with it.

Re: CLI, I'm not sure, I don't think we've discussed that much, but if it's mainly for debugging I think it's fine to use subcommands on the same binary (that's what we do in Contour, FWIW).